monitor/util/metrics_window.go

```diff
@@ -0,0 +1,88 @@
+// Package util provides common functionality for monitoring components.
+package util
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/ipfs/ipfs-cluster/api"
+)
+
+// ErrNoMetrics is returned when there are no metrics in a MetricsWindow.
+var ErrNoMetrics = errors.New("no metrics have been added to this window")
+
+// MetricsWindow implements a circular queue to store metrics.
+type MetricsWindow struct {
+	last int
+
+	safe       bool
```

Comment:
In what case do we want unsafe access?

Reply:
in fact, always, since metrics window is only written when the outer wrappers (MetricStore now) are locked, so no need to double locking.